D/DynarePreprocessor/build_tarballs.jl

```diff
@@ -0,0 +1,69 @@
+using BinaryBuilder
+
+env = copy(ENV)
+env["BINARYBUILDER_AUTOMATIC_APPLE"] = "true"
+env["BINARYBUILDER_USE_CCACHE"] = "true"
+
+name = "DynarePreprocessor"
+version = v"4.8.0-5"
+sha="a8fce06dc46ca09329e6899e1fde47f2cd81809b"
+sources = [
+    GitSource("https://git.dynare.org/Dynare/preprocessor.git", "a8fce06dc46ca09329e6899e1fde47f2cd81809b"),
+]
+
+script = raw"""
+if [[ ${target} == *-apple-* ]]
+then
+    sha="a8fce06dc46ca09329e6899e1fde47f2cd81809b"
+    gitlabjob="37258"
+    wget -O artifacts.zip https://git.dynare.org/Dynare/preprocessor/-/jobs/${gitlabjob}/artifacts/download
+    mkdir work
+    unzip artifacts.zip -d work
+    tar zxf work/${sha}/macos-x86_64/dynare-preprocessor.tar.gz
+    mkdir -p ../destdir/
+    mkdir -p ../destdir/bin
+    cp -r dynare-preprocessor ../destdir/bin
+    install_license ${WORKSPACE}/srcdir/preprocessor/COPYING
+else 
+    apk add samurai
+    apk add boost-dev
+    apk add bison=3.7.6-r0 --update-cache --repository http://dl-cdn.alpinelinux.org/alpine/edge/main 
+    apk add flex-dev
```

Comment:
These are Alpine packages necessary for the build:
- boost-dev C++ header files for Boost
- the grammar for the Dynare modeling language is provided in Flex and Bison. Flex and Bison packages write C++ files from the specification in DynareFlex.ll and DynareBison.yy

Reply:
But why do you need them?  Are they needed for the host or the target system?  The link above explains the difference between the two classes of dependencies.  In any case, we have all of these packages as JLLs, and it'd be better to use them as [`Dependency`](https://docs.binarybuilder.org/stable/reference/#BinaryBuilderBase.Dependency) or [`HostBuildDependency`](https://docs.binarybuilder.org/stable/reference/#BinaryBuilderBase.HostBuildDependency), depending on what they are

Reply:
Because Dynare preprocessor source files contains DynareFlex.ll and DynareBison.yy that are the original source files and don't distribute DynareFlex.cc and DynareBison.cc. So the build system needs the bison and flex utilities to handle these two files. If I understand, I can replace
```
apk add boost-dev
apk add bison=3.7.6-r0 --update-cache --repository http://dl-cdn.alpinelinux.org/alpine/edge/main 
apk add flex-dev
```
with
```
HostBuildDependency("boost-dev")
HostBuildDependency("bison")
HostBuildDependency("flex-dev")
```
?

Reply:
Not quite, you need to use the name of the JLL packages, e.g. flex_jll

Reply:
The others are boost_jll and Bison_jll (note the uppercase start). The bison JLL is quite old, though, predates JLLWrapper, we may want to rebuild it. It's also outdated: 3.5.2, current is 3.8.2

Reply:
See https://github.com/JuliaPackaging/Yggdrasil/pull/4141